### PR TITLE
[Kotlin Cleanup] JSONArrayTest

### PR DESCRIPTION
**Pull request by oyeraghib**
## Purpose / Description
[Kotlin Cleanup] JSONArrayTest

## Checklist

- [ ] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [ ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ ] You have commented your code, particularly in hard-to-understand areas
- [ ] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/utils/JSONArrayTest.kt [hunk 1]
```diff
@@ -39,8 +39,6 @@ import kotlin.test.assertNull
  */
 @RunWith(AndroidJUnit4::class)
 @Config(application = EmptyApplication::class)
-@KotlinCleanup("Use Kotlin's function instead of Arrays.asList")
-@KotlinCleanup("rename variables named object")
 @KotlinCleanup("have toJSONObject return non-null")
```
Comment (oyeraghib) [inline AnkiDroid/src/test/java/com/ichi2/utils/JSONArrayTest.kt:42]: what has to be done for this cleanup ?

File: AnkiDroid/src/test/java/com/ichi2/utils/JSONArrayTest.kt [hunk 2]
```diff
@@ -329,12 +327,12 @@ class JSONArrayTest {
         val values = JSONArray()
         values.put(5.5)
         values.put(false)
-        val `object` = values.toJSONObject(keys)
-        assertEquals(5.5, `object`!!["a"])
-        assertEquals(false, `object`["b"])
+        val value = values.toJSONObject(keys)
+        assertEquals(5.5, value!!["a"])
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/utils/JSONArrayTest.kt:331]: have toJSONObject return non-null means changing the signature of that other method to return  non-null so then lines like this don't need the `!!` - as with all things there is an assumed ", if possible" - by which I mean it should be possible? but maybe when you try it, it turns into a serious mess and seems impractical, only way to know is to try

File: AnkiDroid/src/test/java/com/ichi2/utils/JSONArrayTest.kt [hunk 3]
```diff
@@ -329,12 +326,12 @@ class JSONArrayTest {
         val values = JSONArray()
         values.put(5.5)
         values.put(false)
-        val `object` = values.toJSONObject(keys)
-        assertEquals(5.5, `object`!!["a"])
-        assertEquals(false, `object`["b"])
+        val value = values.toJSONObject(keys)!!
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/utils/JSONArrayTest.kt:329]: Hmm :thinking: - if values.toJSONObjecti is supposed to return non-null per the KotlinCleanup, why do we still need the !! here?

File: AnkiDroid/src/main/java/com/ichi2/utils/JSONArray.java [hunk 4]
```diff
@@ -434,7 +438,7 @@ public List<String> toStringList(String key) {
      * strings will be coerced to strings. This method returns null if either
      * array is empty.
      */
-    public @Nullable JSONObject toJSONObject(@NonNull JSONArray names) {
+    public JSONObject toJSONObject(@NonNull JSONArray names) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/utils/JSONArray.java:441]: This is nullable, why was it removed?

File: AnkiDroid/src/test/java/com/ichi2/utils/JSONArrayTest.kt [hunk 5]
```diff
@@ -39,9 +39,6 @@ import kotlin.test.assertNull
  */
 @RunWith(AndroidJUnit4::class)
 @Config(application = EmptyApplication::class)
-@KotlinCleanup("Use Kotlin's function instead of Arrays.asList")
-@KotlinCleanup("rename variables named object")
-@KotlinCleanup("have toJSONObject return non-null")
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/JSONArrayTest.kt:44]: This needs code contracts for it to work. Keep it for now.

Comment (oyeraghib) [reply AnkiDroid/src/test/java/com/ichi2/utils/JSONArrayTest.kt:44]: Okay got it!

**Comment by oyeraghib** [comment]
I am adding a new PR for this file.

**Pull request closed without merging**